### PR TITLE
Feat/tomb-swap

### PR DESCRIPTION
**Pull request by Jon-edge**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202032085210721

### CODE REVIEW

**Review by samholmes — CHANGES_REQUESTED** [review]
Some preliminary review comments. Primarily, we should break up the first commit we can review the changes there vs file/identifier renames.

File: src/abi/UNISWAP_V2_WERC20_ABI.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// ABI for wrapped tokens
```
[the remark this answers is not in the document]
Comment (Jon-edge) [reply src/abi/UNISWAP_V2_WERC20_ABI.js]: 8ae3eab183de1783769ed1949ac755311b9dc21d

File: src/swap/defi/uni-v2-based/uniV2Contracts.js [hunk 2]
```diff
@@ -1,46 +1,54 @@
...
 const WFTM_TOKEN_ADDRESS = '0x21be370D5312f44cB42ce377BC9b8a0cEF1A4C83'
-export const wrappedFtmToken: Contract = new ethers.Contract(
-  WFTM_TOKEN_ADDRESS,
-  WRAPPED_ERC20_ABI,
-  provider
-)
-
-export const makeErc20Contract = (tokenAddress: string) =>
-  new ethers.Contract(tokenAddress, UNISWAP_V2_ERC20_ABI, provider)
+export const wFtmTokenContract: Contract = (provider: JsonRpcProvider) =>
```
[the remark this answers is not in the document]
Comment (Jon-edge) [reply src/swap/defi/uni-v2-based/uniV2Contracts.js]: f10704f96943a497b26b18c28ba1f50755b7013e

I personally don't mind long names, and find value in having type identifying information in var names. Prevents ambiguity at a glance anywhere.

File: src/swap/spookySwap/spookySwap.js [hunk 3]
```diff
@@ -35,36 +34,33 @@ const swapInfo: EdgeSwapInfo = {
...
+    metaTokens
+  } = fromWallet.currencyInfo
   const fromAddress = (await fromWallet.getReceiveAddress()).publicAddress
 
-  // Sanity check: Both wallets should be of the same chain.
-  if (
-    fromWallet.currencyInfo.currencyCode !== toWallet.currencyInfo.currencyCode
-  )
-    throw new Error('SpookySwap: Mismatched wallet chain')
-
```
Comment (samholmes) [inline src/swap/spookySwap/spookySwap.js:63]: Can we swap between different EVM accounts?

File: src/swap/spookySwap/spookySwap.js [hunk 4]
```diff
@@ -19,28 +19,26 @@ import {
...
   supportEmail: '',
   supportUrl: 'https://discord.com/invite/weXbvPAH4Q'
 }
-const EXPIRATION_MS = 1000 * 20 * 60
-const SLIPPAGE = '0.05' // 5%
 
 export function makeSpookySwapPlugin(
   opts: EdgeCorePluginOptions
 ): EdgeSwapPlugin {
-  const { log } = opts
```
Comment (samholmes) [inline src/swap/spookySwap/spookySwap.js:187]: Hmm, logs are useful for support help

File: src/swap/defi/uni-v2-based/uniV2Contracts.js [hunk 5]
```diff
@@ -19,13 +19,22 @@ export const provider: Provider = new ethers.providers.JsonRpcProvider(
...
 )
 
+const TOMBSWAP_ROUTER_ADDRESS = '0x6D0176C5ea1e44b08D3dd001b0784cE42F47a3A7'
+export const tombSwapRouter: Contract = new ethers.Contract(
+  TOMBSWAP_ROUTER_ADDRESS,
+  UNISWAP_V2_ROUTER_ABI,
+  provider
+)
+
+// Wrapped Tokens
```
Comment (samholmes) [inline src/swap/defi/uni-v2-based/uniV2Contracts.js:46]: Couldn't we just have a factory function to make this contract type given an address? Then we could parameterize the plugin with either the address or the contract returned from the factory (similar to how we do it in stake-plugins).

File: src/swap/defi/uni-v2-based/plugins/tombSwap.js [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+const SLIPPAGE = '0.05'
+
+const swapInfo: EdgeSwapInfo = {
+  pluginId: 'tombSwap',
+  displayName: 'TombSwap',
+  supportEmail: '',
+  supportUrl: 'https://discord.gg/vANnESmVdz'
+}
+
+export function makeTombSwapPlugin(
```
Comment (samholmes) [inline src/swap/defi/uni-v2-based/plugins/tombSwap.js:35]: If the implementation is nearly the same as `makeSpookySwapPlugin` then why not `makeUniswapPlugin` with parameters in the plugin info which determine the contract address to use, etc?

File: src/swap/defi/uni-v2-based/uniV2Contracts.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (samholmes) [reply src/swap/defi/uni-v2-based/uniV2Contracts.js]: I agree

File: src/swap/defi/uni-v2-based/uniV2Utils.js [hunk 7]
```diff
@@ -46,6 +46,7 @@ export const getSwapAmounts = async (
  * Get smart contract transaction(s) necessary to swap based on swap params
  */
 export const getSwapTransactions = async (
+  provider: JsonRpcProvider,
```
Comment (samholmes) [inline src/swap/defi/uni-v2-based/uniV2Utils.js:49]: These functions feel like methods to the swap plugin rather than utils; don't you think?

**Review by samholmes — CHANGES_REQUESTED** [review]

File: src/swap/defi/uni-v2-based/plugins/tombSwap.js (hunk 6 above)
Comment (samholmes) [reply src/swap/defi/uni-v2-based/plugins/tombSwap.js:35]: I'll secede this request for a later refactor.

File: src/swap/defi/uni-v2-based/uniV2Utils.js (hunk 7 above)
Comment (samholmes) [reply src/swap/defi/uni-v2-based/uniV2Utils.js:49]: I'll secede this request for a later refactor.

**Review by thehobbit85 — APPROVED** [review]

**Pull request merged**